apps/api-extractor/src/generators/ApiModelGenerator.ts

```diff
@@ -0,0 +1,369 @@
+  private readonly _collector: Collector;
+  private readonly _cachedOverloadIndexesByDeclaration: Map<AstDeclaration, number>;
+  private readonly _apiModel: ApiModel;
+
+  public constructor(collector: Collector) {
+    this._collector = collector;
+    this._cachedOverloadIndexesByDeclaration = new Map<AstDeclaration, number>();
+    this._apiModel = new ApiModel();
+  }
+
+  public get apiModel(): ApiModel {
+    return this._apiModel;
+  }
+
+  public buildApiPackage(): ApiPackage {
+    const packageDocComment: tsdoc.DocComment | undefined = this._collector.package.tsdocComment;
+
+    const apiPackage: ApiPackage = new ApiPackage({
+      name: this._collector.package.name,
+      docComment: packageDocComment
+    });
+    this._apiModel.addMember(apiPackage);
+
+    const apiEntryPoint: ApiEntryPoint = new ApiEntryPoint({ name: '' });
+    apiPackage.addMember(apiEntryPoint);
+
+    // Create a CollectorEntity for each top-level export
+    for (const entity of this._collector.entities) {
+      for (const astDeclaration of entity.astSymbol.astDeclarations) {
+        if (entity.exported) {
+          this._processDeclaration(astDeclaration, entity.nameForEmit, apiEntryPoint);
+        }
+      }
+    }
+
+    return apiPackage;
+  }
+
+  private _processDeclaration(astDeclaration: AstDeclaration, exportedName: string | undefined,
+    parentApiItem: ApiItemContainerMixin): void {
+
+    if ((astDeclaration.modifierFlags & ts.ModifierFlags.Private) !== 0) {
+      return; // trim out private declarations
+    }
+
+    switch (astDeclaration.declaration.kind) {
+      case ts.SyntaxKind.ClassDeclaration:
+        this._processApiClass(astDeclaration, exportedName, parentApiItem);
+        break;
+
+      case ts.SyntaxKind.InterfaceDeclaration:
+        this._processApiInterface(astDeclaration, exportedName, parentApiItem);
+        break;
+
+      case ts.SyntaxKind.MethodDeclaration:
+        this._processApiMethod(astDeclaration, exportedName, parentApiItem);
+        break;
+
+      case ts.SyntaxKind.MethodSignature:
+        this._processApiMethodSignature(astDeclaration, exportedName, parentApiItem);
+        break;
+
+      case ts.SyntaxKind.ModuleDeclaration:
+        this._processApiNamespace(astDeclaration, exportedName, parentApiItem);
+        break;
+
+      case ts.SyntaxKind.PropertyDeclaration:
+        this._processApiProperty(astDeclaration, exportedName, parentApiItem);
+        break;
+
+      case ts.SyntaxKind.PropertySignature:
+        this._processApiPropertySignature(astDeclaration, exportedName, parentApiItem);
+        break;
+
+      case ts.SyntaxKind.Constructor:
+      case ts.SyntaxKind.ConstructSignature:
+      case ts.SyntaxKind.EnumDeclaration:
+      case ts.SyntaxKind.EnumMember:
+      case ts.SyntaxKind.FunctionDeclaration:
+      case ts.SyntaxKind.IndexSignature:
+      case ts.SyntaxKind.TypeAliasDeclaration:
+      case ts.SyntaxKind.VariableDeclaration:
+        default:
```

Comment:
>default [](start = 8, length = 7)

Decrease indent.

Reply:
Indentation seems correct to me.

---
In reply to: [236471702](https://github.com/Microsoft/web-build-tools/pull/952#discussion_r236471702) [](ancestors = 236471702)